### PR TITLE
Add tests for `ParametrizedCurve`

### PR DESCRIPTION
**Pull request by JoshuaLampert**
I added tests for the new `ParametrizedCurve` in Meshes.jl. I used a curve defined in Cartesian and one in Polar coordinates to test if different crs work as expected. I also added it to the support matrix and bumped the minimum compat version for Meshes.jl in the tests accordingly. Since I defined the curve to describe a 2D sphere for simplicity, this is also used as an analytical solution for the 2D `Sphere` (see #67).

### CODE REVIEW

**Comment by JoshuaLampert** [comment]
Hm, downgrade complains because the tests need Meshes.jl v0.51.20 because this is the first version containing `ParametrizedCurve`, but the compat for Meshes.jl in MeshIntegrals.jl itself is `0.50, 0.51`, thus allowing 0.50. But we don't need to bump the lower compat for Meshes.jl in MeshIntegrals.jl because `ParametrizedCurve` is never used in the source. I'm not sure how to deal with this problem.

**Comment by mikeingold** [comment]
Is it possible to make this test item on a separate environment with a specified minimum version?

**Comment by mikeingold** [comment]
Or maybe some kind of conditional statement on the installed version?

**Comment by JoshuaLampert** [comment]
What we could is to define an environment variable in the CI setup, which is, e.g. equal to "part1" or "part2" and split up the tests into two parts. Let's say the new test item is in part2. Then we could run the Downgrade action only for part1 and all the other CI actions run on both "part1" and "part2".

**Comment by JoshuaLampert** [comment]
This way we would also reduce the CI run time.

**Comment by JoshuaLampert** [comment]
> Or maybe some kind of conditional statement on the installed version?

This could maybe also work.

**Comment by JoshuaLampert** [comment]
> What we could is to define an environment variable in the CI setup, which is, e.g. equal to "part1" or "part2" and split up the tests into two parts. Let's say the new test item is in part2. Then we could run the Downgrade action only for part1 and all the other CI actions run on both "part1" and "part2".

Meshes.jl does something similar for Float32 and Float64 tests.

**Comment by JoshuaLampert** [comment]
>> Or maybe some kind of conditional statement on the installed version?

> This could maybe also work.

There is  `pkgversion` for that. This seems like a simple solution. Let's see if tests pass now.

File: test/combinations.jl [hunk 1]
```diff
@@ -703,13 +750,18 @@ end
...
+    function f(p::P) where {P <: Meshes.Point}
+        ur = hypot(p.coords.x, p.coords.y)
+        r = ustrip(u"m", ur)
+        exp(-r^2)
+    end
     fv(p) = fill(f(p), 3)
 
     # Scalar integrand
-    sol = Meshes.measure(sphere)
+    sol = 2π * radius * exp(-radius^2) * u"m"
```
Comment (mikeingold) [inline test/combinations.jl:764]: Nice!

File: test/combinations.jl [hunk 2]
```diff
@@ -516,6 +516,53 @@ end
...
 end
 
+@testitem "ParametrizedCurve" setup=[Setup] begin
+    # ParametrizedCurve has been added in Meshes v0.51.20
+    # If the version is specified as minimal compat bound in the Project.toml, the downgrade test fails
+    if pkgversion(Meshes) >= v"0.51.20"
+        using CoordRefSystems: Polar
+        using LinearAlgebra: norm
+
+        radius = 4.4
```
Comment (mikeingold) [inline test/combinations.jl:526]:
```suggestion
        # Parameterize a circle centered on origin with specified radius
        radius = 4.4
```

Comment (mikeingold) [reply test/combinations.jl:526]: I think adding a comment here might be useful to help us remember what this geometry is modeling.

**Pull request merged**